### ISSUE TITLE
Generalize this class of authority

### ISSUE THREAD

**Issue by nmushegian** [ethpm/ethereum-whitelist-authority#1]
I think this is the first custom Authority I saw in the wild, yay!

The latest [ds-whitelist](https://github.com/dapphub/ds-whitelist) has this `canCall`:

```
return _isRoot[caller]
    || _isPublicCode[code]
    || _isPublicCapability[code][sig]
    || _canCall[caller][code][sig];
```

I think this is a generalization of "BasicAuthority", old DSWhitelist, and your Whitelist

Is that right? Should we merge these projects?

(The example we have for an authority type that *isn't* generalized by ds-whitelist is [ds-roles](https://github.com/dapphub/ds-roles) )

**Comment by pipermerriam** [comment]
Yes, that does indeed seem to be a superset of this contract's functionality.  I'll happily use that package if you guys want to publish it.

**Comment by pipermerriam** [comment]
> I think this is the first custom Authority I saw in the wild, yay!

Also, this model has grown on me a *lot* since I first saw it wayyyy back when I audited dappsys.
